### ISSUE TITLE
allow remote config configuration inline in a command

### ISSUE THREAD

**Issue by xenoterracide** [hashicorp/terraform#12464]
Feature request/suggestion. writing this for a `Dockerfile` has actually been kind of annoying, and there's the problem of it's not really clear whether or if using remote config if you can create the s3 bucket (or other), with terraform.

I'm suggesting that instead of doing this (note in both cases `$@` could contain `apply` or some other command.
```
#!/bin/sh
set -x
/bin/terraform -version
/bin/terraform remote config -backend=s3 -backend-config="bucket=com_xenoterracide_terraform_prod" -backend-config="key=iam/terraform.tfstate" -backend-config="region=us-east-1"
/bin/terraform "$@"
```

allow
```
#!/bin/sh
set -x
/bin/terraform -version
/bin/terraform -backend=s3 -backend-config="bucket=com_mydomain_terraform_prod" -backend-config="key=iam/terraform.tfstate" -backend-config="region=us-east-1" "$@"
```

this shouldn't fail on retrieval if this bucket is missing as it could be used to create the bucket. 

that said it should probably error and fall back to a local save is persist fails.

**Comment by SydOps** [comment]
since 0.9 is released,` remote` command will not be supported after 0.11.

But I will be happy if `terraform init` can support this 

>/bin/terraform init -backend=s3 -backend-config="bucket=com_mydomain_terraform_prod" -backend-config="key=iam/terraform.tfstate" -backend-config="region=us-east-1" 

Refer: https://github.com/hashicorp/terraform/pull/12067

**Comment by xenoterracide** [comment]
yeah but I want it in the `apply`, etc, commands, so when doing Continuous Deployment of provisioning infrastructure, you don't have to run 2 commands, this is especially true of the current docker container in which running 2 commands would be painful without extension because of its entrypoint.

**Comment by SydOps** [comment]
@xenoterracide 

`terraform remote` is gone in 0.9.0 already. Currently you have to run `terraform init` first, then run `apply`

**Comment by xenoterracide** [comment]
I'm sure that's an improvement somehow... but as far as this running of multiple commands thing in CI it isn't really.... I don't understand why we can't just have an init config in repo.. and then just run `terraform apply` without running init

**Comment by SydOps** [comment]
@xenoterracide 

There are some explanation in #12067, that the owner of terraform do that on purpose. You can follow it for updates.
